### PR TITLE
[Deprecated][inductor] switch CppCodeCache to new cpp_builder.

### PR DESCRIPTION
**Pull request by xuhancn**
This PR is the implemention of https://github.com/pytorch/pytorch/issues/124245 step 2.1
1. Switch `CppCodeCache` to new CPP builder.
2. Switch `AotCodeCompiler` to new CPP builder.
3. Switch `HalideCodeCache` to new CPP builder.

TODO:
Current still need to optimize code structure and circle import. I will do this work in further PR. I have did the exploration: https://github.com/pytorch/pytorch/pull/128330  

cc @peterjc123 @mszhanyi @skyline75489 @nbcsm @vladimir-aubrecht @iremyux @Blackhex @cristianPanaite @jgong5 @mingfeima @XiaobingSuper @sanchitintel @ashokei @jingxu10 @voznesenskym @penguinwu @EikanWang @Guobing-Chen @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @peterbell10 @ipiszy @yf225 @chenyang78 @kadeng @muchulee8 @ColinPeppler @amjames @desertfire @chauhang

### CODE REVIEW

**Comment by xuhancn** [comment]
Hi @desertfire ,
This PR is fully switch codecache to new cpp_builder, could you please help on import to Meta fb_code environment? Much appreciated!

**Comment by desertfire** [comment]
internal-external out of sync. Will need to wait for a while.

**Comment by xuhancn** [comment]
Hi @desertfire , could you please let me know the import status?

File: torch/_inductor/codecache.py [hunk 1]
```diff
@@ -1985,15 +1985,29 @@ def compile(
...
-        cpp_command = repr(
-            cpp_compile_command(
-                "i",
-                "o",
+
+        from torch._inductor.cpp_builder import (
+            CppBuilder,
+            CppTorchCudaOptions,
+            get_name_and_dir_from_output_file_path,
+        )
```
Comment (jgong5) [inline torch/_inductor/codecache.py:1993]: I feel we need some refactoring to avoid the cyclic dep here...

**Review by jgong5 — DISMISSED** [review]
Can you remove the unused or commented out code?

File: torch/_inductor/codecache.py (hunk 1 above)
Comment (xuhancn) [reply torch/_inductor/codecache.py:1993]: @jgong5 Plan to restructure code to solve it in next PR, I have did the exploration: https://github.com/pytorch/pytorch/pull/128330

**Review by jgong5 — APPROVED** [review]

**Comment by xuhancn** [comment]
@desertfire this PR seems damaged, please move to new one: https://github.com/pytorch/pytorch/pull/129144

**Pull request closed without merging**